library/HTMLPurifier/HTMLModule/HTML5/SafeExternalStylesheets.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/*
+ * https://html.spec.whatwg.org/dev/semantics.html#the-link-element
+ */
+class HTMLPurifier_HTMLModule_HTML5_SafeExternalStylesheets extends HTMLPurifier_HTMLModule
+{
+    /**
+     * @type string
+     */
+    public $name = 'HTML5_SafeExternalStylesheets';
+
+    /**
+     * @param HTMLPurifier_Config $config
+     */
+    public function setup($config)
+    {
+        $allowed = $config->get('HTML.SafeExternalStylesheet');
+
+        // https://html.spec.whatwg.org/dev/semantics.html#the-link-element
+        $this->addElement('link', 'Flow', 'Empty', null, array(
+            'rel*'  => new HTMLPurifier_AttrDef_Enum(array('stylesheet')),
+            'type'  => new HTMLPurifier_AttrDef_Enum(array('text/css')),
```

Comment:
> We also need LinkType attribute definition (with predefined list of allowed values, similar to ARel), and a LinkRel attribute definition as well (I assume valid rel values differ between a and link, so we need a separate definition class)

I'll need to check if it's just a case of adding these values to the enum, or if there's attributes they require implementing. I was just focusing on stylesheet support.
> The body-ok keywords are dns-prefetch, modulepreload, pingback, preconnect, prefetch, preload, prerender, and stylesheet.